std/traits.d

```diff
@@ -6524,6 +6575,40 @@ unittest
         static assert(mostNegative!T == 0);
 }
 
+/**
+Get the type that a scalar type `T` will promote to in arithmetic expressions.
```

Comment:
Link (I will add it to the PR, as well): https://dlang.org/spec/type.html#integer-promotions

For the purposes of my `Promoted` trait, the rules are both simple and sensible:

* Any basic type smaller than `int` is promoted to `int`.

* `dchar` promotes to `uint`.

The problem comes when you mix two different types, like `int` and `uint` - at that point, what happens is the *opposite* of what should happen: the signed type will be promoted to the **un**signed type. This is awful, but required by D's "*C code that compiles in D should do the same thing it did in C*" rule.

The need to warn against implicit signed to unsigned conversions stuff it is what makes `SafeInt` such a pain to use, motivating me to create `SmartInt` which uses better rules.